src/pages/index.js

```diff
@@ -1,275 +1,242 @@
+        })
+        .then(() => {
+          userProfile.setUserImage(inputValues["avatar-link"]);
+          resolve();
+        })
+        .catch(() => reject());
+    });
+  },
+});
 
-//#region preview modal declarations
-const previewModal = document.querySelector('#preview-modal');
-const previewImage = document.querySelector('.modal__preview-image');
-const previewTitle = document.querySelector('#preview-title');
-previewModal.addEventListener('click',checkForModalOverlayClick);
-document.querySelector('#preview-modal__close').addEventListener('click',onCloseModalClick);
-//#endregion
+const editAvatarFormValidator = new FormValidator(
+  validationSettings,
+  editAvatarForm.getFormElement()
+);
+editAvatarFormValidator.enableValidation();
+
+const editProfileForm = new ModalWithForm({
+  modalSelector: modalSelectors.editProfileModal,
+  onSubmitCallback: (inputValues) => {
+    return new Promise((resolve, reject) => {
+      dbApi
+        .request(routeData.updateUserProfile, {
+          name: inputValues["profile-name"],
+          about: inputValues["profile-description"],
+        })
+        .then(() => {
+          userProfile.setUserName(inputValues["profile-name"]);
+          userProfile.setUserDescription(inputValues["profile-description"]);
+          editProfileForm.setDefaultInputs(userProfile.getUserProfile());
+          resolve();
+        })
+        .catch((err) => {
+          console.error(`Failed to edit profile: ${err}`);
+          reject();
+        });
+    });
+  },
+});
 
-//#region modal handlers
-function checkForModalOverlayClick(evt) {
-    if(evt.target.classList.contains("modal")) onCloseModalClick(evt);
-}
+const editProfileFormValidator = new FormValidator(
+  validationSettings,
+  editProfileForm.getFormElement()
+);
+editProfileFormValidator.enableValidation();
+
+const newPostForm = new ModalWithForm({
+  modalSelector: "#post-modal",
+  onSubmitCallback: (inputValues) => {
+    return new Promise((resolve, reject) => {
+      const newPostTitle = inputValues["post-caption"];
+      const newPostImageLink = inputValues["post-image-link"];
+      dbApi
+        .request(routeData.createCard, {
+          name: newPostTitle,
+          link: newPostImageLink,
+        })
+        .then((data) => {
+          cardSection.createAndRenderItem(data);
+          resolve();
+        })
+        .catch((err) => {
+          console.error(`Failed to create new post: ${err}`);
+          reject();
+        });
+    });
+  },
+});
 
-function onCloseModalClick(evt) {
-    const modal = evt.target.closest('.modal');
-    closeModal(modal);
-}
+newPostForm.setPlaceholderInputs(newPostModalPlaceholders);
 
-function openModal(modal) {
-    addEscapeListener();
-    // checkFormFieldInputErrors(modal.querySelector('.modal__form')); //this can check every field on open, but feels bearish
-    modal.classList.add('modal_open');
-}
+const newPostFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```

Reply:
Sooooo much tidier. Thank you!